src/System.Private.CoreLib/shared/System/Reflection/AssemblyName.cs

```diff
@@ -144,19 +131,20 @@ public AssemblyContentType ContentType
         // Make a copy of this assembly name.
         public object Clone()
         {
-            AssemblyName name = new AssemblyName();
-            name.Init(_name,
-                      _publicKey,
-                      _publicKeyToken,
-                      _version,
-                      _cultureInfo,
-                      _hashAlgorithm,
-                      _versionCompatibility,
-                      _codeBase,
-                      _flags,
-                      _strongNameKeyPair);
-            name._hashForControl = _hashForControl;
-            name._hashAlgorithmForControl = _hashAlgorithmForControl;
+            var name = new AssemblyName
+            {
+                _name = _name,
+                _publicKey = (byte[])_publicKey?.Clone(),
```

Comment:
This will clone the array twice for no good reason - once here and second time in the constructor.

Reply:
@jkotas um.. I don't get it - this Clone uses object initializing syntax `new AssemblyName {` where is the second array clone happening?

Reply:
Ok, the clone is fine. But why you need the cloning in the internal constructor? Is there a path where the internal constructor is called with arrays that are shared with something else?

Reply:
Yeah looks like the clone calls are not needed there. fixed